File: LeetCode/3217-delete-nodes-from-linked-list-present-in-array/3217-delete-nodes-from-linked-list-present-in-array.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {
        set<int> st(nums.begin(), nums.end());
        while (head && st.find(head->val) != st.end()) {
            ListNode* toDelete = head;
            head = head->next;
            delete toDelete;
        }
        
        if (!head) {
            return nullptr;
        }
        
        ListNode *prev = head;
        ListNode *temp = head->next;
        
        while (temp) {
            if (st.find(temp->val) != st.end()) {
                prev->next = temp->next;
                ListNode* toDelete = temp;
                temp = temp->next;
                delete toDelete;
            } else {
                prev = temp;
                temp = temp->next;
            }
        }
        
        return head;
    }
};

File: LeetCode/3217-delete-nodes-from-linked-list-present-in-array/3217-delete-nodes-from-linked-list-present-in-array.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        st = set(nums)
        while head and head.val in st:
            head = head.next
        if head is None:
            return None
        prev = head
        temp = head.next

        while temp:
            if temp.val in st:
                prev.next = temp.next
            else:
                prev = temp
            temp = temp.next
        
        return head